Update workflows.adoc (#8722)
circleci/circleci-docs: jekyll/_cci2/workflows.adoc

Workflows may appear with one of the following states:

[.table.table-striped]
[cols=3*, options="header", stripes=even]
|===
| State | Description | Terminal State

| RUNNING
| Workflow is in progress
| No

| NOT RUN
| Workflow was never started

> | No

Reviewer: NOT RUN is a terminal state; there is no other state it can transition to from there.

```suggestion
| Yes
```

| CANCELED
| Workflow was canceled before it finished
| Yes

| FAILING

> | A job in the workflow has failed

Reviewer: Maybe clarify a bit more the difference between this and `failed`

```suggestion
| A job in the workflow has failed, but others are still running or yet to be approved
```

| No

| FAILED
| One or more jobs in the workflow failed
| Yes

| SUCCESS
| All jobs in the workflow completed successfully
| Yes

> | NEEDS APPROVAL

Reviewer: ~ just a note that in the backend and APIs this state is `ON_HOLD`, but the UI started renaming it as of 2 months ago. Should we include both statuses? Or a separate column for the UI status?

```suggestion
| NEEDS APPROVAL (UI) / ON HOLD
```

| A job in the workflow is waiting for approval
| No

> | NEEDS SETUP
> | A workflow stanza is not included or is incorrect in the xref:configuration-reference#[`.circleci/config.yml` file] for this project
> | No
> |===

Reviewer: `NEEDS SETUP` is not actually a workflow state we support. From the description I think what is meant here is `ERROR`, which is a terminal status. I've searched the UI to verify, and it does not look like this is a case of the UI renaming the `ERROR` state to `NEEDS SETUP` (like it does for `ON HOLD` -> `NEEDS APPROVAL`)

```suggestion
| ERROR
| A workflow stanza is not included or is incorrect in the xref:configuration-reference#[`.circleci/config.yml` file] for this project
| Yes

|
| UNAUTHORIZED
| One or more of the jobs terminated with a `unauthorized` job status, because the triggerer does not have access to a required restricted context.
| Yes
|===
```

> NOTE: After 90 days non-terminal workflows will be cancelled by the CircleCI Automation actor

Reviewer: ```suggestion
NOTE: After 90 days non-terminal workflows will be cancelled by the CircleCI Automation actor.
```

Workflows may appear with one of the following states:

[.table.table-striped]
[cols=3*, options="header", stripes=even]
|===

> | State | Description | Terminal State

Reviewer 2: ```suggestion
| State | Description | Terminal state
```

| RUNNING
| Workflow is in progress
| No

| NOT RUN
| Workflow was never started
| Yes

| CANCELED
| Workflow was canceled before it finished
| Yes

| FAILING
| A job in the workflow has failed, but others are still running or yet to be approved
| No

| FAILED
| One or more jobs in the workflow failed
| Yes

| SUCCESS
| All jobs in the workflow completed successfully
| Yes

| NEEDS APPROVAL (UI) / ON HOLD
| A job in the workflow is waiting for approval
| No

| ERROR

> | A workflow stanza is not included or is incorrect in the xref:configuration-reference#[`.circleci/config.yml` file] for this project

Reviewer: It occurred to me that config issues are now caught higher up the stack. We'll get an error at the pipeline level if there are issues with the validity of the config. I had another look at this and the only time we set an `ERROR` status for workflows is when give up processing an `unblock-jobs` message after 10 minutes of retries, which in the majority cases is due to a failure building the permissions profile. It is quite rare.

```suggestion
| We experienced an internal error starting a job in the workflow
```
